adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/FeatureRDD.scala

```diff
@@ -63,19 +63,85 @@ trait FeatureOrdering[T <: Feature] extends Ordering[T] {
       .result()
   }
 }
-object FeatureOrdering extends FeatureOrdering[Feature] {}
+private object FeatureOrdering extends FeatureOrdering[Feature] {}
 
-class FeatureRDDFunctions(featureRDD: RDD[Feature]) extends Serializable with Logging {
+object FeatureRDD {
 
+  /**
+   * @param elem The feature to extract a sequence record from.
+   * @return Gets the SequenceRecord for this feature.
+   */
+  private def getSequenceRecord(elem: Feature): SequenceRecord = {
+    SequenceRecord(elem.getContigName, 1L)
+  }
+
+  /**
+   * Builds a FeatureRDD without SequenceDictionary information by running an
+   * aggregate to rebuild the SequenceDictionary.
+   *
+   * @param rdd The underlying Feature RDD to build from.
+   * @return Returns a new FeatureRDD.
+   */
+  private[rdd] def apply(rdd: RDD[Feature]): FeatureRDD = {
+
+    // cache the rdd, since we're making multiple passes
+    rdd.cache()
+
+    // aggregate to create the sequence dictionary
+    val sd = new SequenceDictionary(rdd.map(getSequenceRecord)
+      .distinct
+      .collect
+      .toVector)
+
+    FeatureRDD(rdd, sd)
+  }
+}
+
+case class FeatureRDD(rdd: RDD[Feature],
+                      sequences: SequenceDictionary) extends AvroGenomicRDD[Feature, FeatureRDD] {
+
+  /**
+   * Java friendly save function. Automatically detects the output format.
+   *
+   * If the filename ends in ".bed", we write a BED file. If the file name ends
```

Comment:
To be clear, these actually write a directory full of `part...` files in the various feature formats, since we're using Spark `saveAsTextFile`.  Support for single files is #1058.